test/tests/functional/pbs_passing_environment_variable.py

```diff
@@ -74,14 +74,17 @@ def test_commas_in_custom_variable(self):
              'Resource_List.walltime': 10}
         script = ['#PBS -v "var1=\'A,B,C,D\'"']
         script += ['env | grep var1']
-        jid = self.create_and_submit_job(content=script)
+        jid = self.create_and_submit_job(user=TEST_USER, content=script)
         qstat = self.server.status(JOB, ATTR_o, id=jid)
         job_outfile = qstat[0][ATTR_o].split(':')[1]
 
         self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=10)
         job_output = ""
-        with open(job_outfile, 'r') as f:
-            job_output = f.read().strip()
+        runcmd = ['cat', job_outfile]
+        ret = self.du.run_cmd(self.server.client, runcmd,
+                              level=logging.INFOCLI, runas=TEST_USER,
+                              logerr=False)
+        job_output =(' '.join(map(str, ret['out']))).strip()
```

Comment:
I believe the output of ret['out'] is a list of strings.  I don't think you need to map it through str.  In any case, the output of self.du.cat() is a list of strings, so it will work.

Reply:
I'm still not sure you need to do the map().  I thought the output of self.du.cat( ) was a list of strings.  You should be able to do the join() directly.

Reply:
@bhroam i have remove the map().